File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>firstOccur;
        int n=nums.size(), prefix = 0;
        firstOccur[0] = -1;
        for(int i=0; i<n;i++){
            prefix+=nums[i];
            if(firstOccur.find(prefix%k) != firstOccur.end()){
                 if(i-firstOccur[prefix%k] >= 2) return true;
            }
            else{
                firstOccur[prefix % k] = i;
            }
      }
       return false;

    }      
    
};

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        int n = nums.length, prefix = 0;
        HashMap<Integer,Integer> firstOcc = new HashMap<>();
        firstOcc.put(0,-1);
        for(int i=0; i<n; i++){
            prefix += nums[i];

            if(firstOcc.containsKey(prefix % k)){
                if(i-firstOcc.get(prefix%k) >= 2) return true;
            }
            else{
                firstOcc.put(prefix%k , i);
            }
        }
        return false;
    }
}